pydicom/dataelem.py

```diff
@@ -865,27 +885,133 @@ def DataElement_from_raw(
         )
         if config.convert_wrong_length_to_UN:
             warnings.warn(f"{message} Setting VR to 'UN'.")
-            vr = VR_.UN
+            raw = raw._replace(VR=VR_.UN)
             value = raw.value
         else:
             raise BytesLengthException(
                 f"{message} To replace this error with a warning set "
                 "pydicom.config.convert_wrong_length_to_UN = True."
             )
+    return raw, {"value": value}
+
+
+def raw_LUT_descriptor_handler(
+    raw: RawDataElement,
+    **kwargs,
+) -> Tuple[RawDataElement, dict]:
+    """Handle post-convert_value processing of a :class:`RawDataElement`.
 
-    if raw.tag in _LUT_DESCRIPTOR_TAGS and value:
+    Parameters
+    ----------
+    raw_data_element : RawDataElement
+        The raw data element with the VR and value to convert.
+
+    Returns
+    -------
+    RawDataElement
+        The raw data element
+    dict
+        Any kwargs to pass into downstream handlers.
+    """
+    if raw.tag in _LUT_DESCRIPTOR_TAGS and raw.value:
+        value = kwargs.get('value', [])
         # We only fix the first value as the third value is 8 or 16
         try:
             if value[0] < 0:
                 value[0] += 65536
         except TypeError:
             pass
+        return raw, {'value': value}
+    return raw, {}
+
+
+@functools.lru_cache()
+def import_func(func_path: Union[str, Callable]) -> Callable:
+    """Return function imported from a fully qualified name."""
+    if callable(func_path):
+        return func_path
+    try:
+        module_name, func_name = func_path.rsplit(".", maxsplit=1)
+        module = importlib.import_module(module_name)
+        func = getattr(module, func_name)
+    except (ValueError, ImportError, AttributeError) as exc:
+        msg = f"Cannot import callback {func_path!r} ({exc})"
+        raise ValueError(msg) from exc
+    assert callable(func), f"{func_path} is not callable"
+    return func
+
```

Comment:
Is it necessary to allow users to use a string name?  They can always import themselves and pass a callable, can't they?  If it is to avoid circular import, that shouldn't be a problem if the 'standard' ones are imported within the class, because those aren't executed at config module import.